### ISSUE TITLE
Image: scroll performance issues caused by background-image CSS property

### ISSUE THREAD

**Issue by faceyspacey** [necolas/react-native-web#227]
Because `Image` uses `background-image`, when scrolling lots of images, performance is destroyed. Here's a non-RNW post regarding the issue with this css property in general:

http://stackoverflow.com/questions/33229840/chrome-is-lagging-when-using-css-background-sizecover-in-scrolling-background-i

The `translateZ` hardware-acceleration trick mentioned there hardly does anything in this case. And the issue isn't just about `background-size: cover`. I've created my own `<Image>` class and tried every combination. Just having `background-image` is enough to kill performanec for me. I have 50 images on the page. With under 20 images you won't notice it unless you have something to compare to. If you do try the alternative and create your own `<Image>` class which doesn't use background images, you will definitely notice a major different on just those 10 images.  *Now*, for all 50, the difference is between a page that barely scrolls and one that scrolls properly. It's a major problem--you basically can't use the `<Image>` component that comes with RNW for more than a few images. 

The only option I see is rolling your own that can't properly center an image like you do with `background-size: cover`, or otherwise RNW will be incompatible with the `resizeMode` expectations of standard RN.

**Comment by necolas** [comment]
The joys of the web platform. Thanks for tracking this issue down. Is it an issue in all browsers or just Chrome?

**Comment by necolas** [comment]
FWIW, we use background-image for all media previews on mobile.twitter.com. Perhaps we don't experience this issue because we avoid rendering off-screen timeline items.

**Comment by RangerMauve** [comment]
The stackoverflow link says that doing a 3D transform to move the layer to the GPU is supposed to help. Would that be a good idea to pursue, or should the `background-image` properly just be replaced entirely?

**Comment by necolas** [comment]
Didn't you mention that the 3D transform didn't help? Some of the display modes require background-image to work (e.g., repeat)

**Comment by faceyspacey** [comment]
@RangerMauve I've done the `translateZ` thing and in this case, I couldn't notice anything. If you check the *Paint Flashing* box in Chrome's *Rendering Settings* you will see the paint still flashing when scrolling.  Basically with the `background-image` property, it requires re-paints on scrolling. It seems as simple as that. 

Here's some screenshots of the Rendering Settings in action:

https://www.dropbox.com/s/u4akheugmj331yh/Screenshot%202016-10-19%2014.52.21.png?dl=0

https://www.dropbox.com/s/7n58s4sz0emceso/Screenshot%202016-10-19%2014.52.50.png?dl=0
the green borders around the images show the images are being repainted.  

Here's what I'm using to replicate `background-size: cover;` in javascript as a replacement for the `<Image>` component:

https://gist.github.com/faceyspacey/87d6471974b90978704f40ef522b8bd3
*Using this, there are no green borders indicating re-paints.*

@necolas I haven't tested not using Chrome. But I have noticed my animations are horrible in Firefox using `Animated` and React. I mean, where Chrome renders 100 frames, Firefox is rendering 3-5. That's a whole other issue, and I'm not even sure where to start. I started out using React with React Native a year ago, so I can't speak for plain React. RNW is my only experience with it. The tests I've done in Firefox thus far are this: using `Animated` to animate something from the console so that I'm sure no other renderings are happening, and in Firefox it animates nicely. However in my actual app after renderings just occurred, it's horrible. So it would seem that firefox really struggles to render things and animate things in close succession--and I've done a lot using `shouldComponentUpdate` to insure animations happen without re-rendering and have gotten things super smooth in Chrome. My guess is Firefox is still garbage collecting for a while after renderings happen, but I haven't checked Firefox's equivalent of Chrome's Timeline. Coming from React Native, I decided to build my website like an "application," i.e. highly animated. I'm very concerned about Firefox, and I haven't even begun to look at IE, but perhaps they're closer to Chrome in terms of performance these days.

**Comment by RangerMauve** [comment]
You could try applying the transform on the parent container for the images, that way the might all be placed on one layer for the GPU.

An alternative to this might be some sort of gross canvas hacks, or absolute positioning tricks.

**Comment by faceyspacey** [comment]
@RangerMauve I did apply it to the parent. ..But actually, I just went back to double check, and `<Image>` doesn't pass along the `className` attribute like `<View>` does, so it wasn't applied. Just now I've tested `translate3d(0,0,0)` and `tranzlateZ(0)` and they both have the same result: VERY POSITIVE! It's way better. 

I have an app with a bunch of images on the page all using the Image component I created in the gist--so it's very easy for me to make that component return the standard RNW `Image` component and alternate between the 2 and compare the difference in a real use case with lots of images, which I just did. And if the performance before had a score of 
0 and with the gist, 100, we are now somewhere between 65 and 75. 

So, basically this bug is closed, pending a PR to add `translateZ(0)`. We just need to make sure if developers support a transform style property, it's merged. 

Final Thought: the performance is still *very noticeably* better manually replicating `background-position: cover` in javascript. But I'm also animating my images, which needs the utmost of performance. So for a standard use case of just scrolling non-animated images, `translateZ(0)` should suffice. Hopefully this little discovery brings what could have been a big upset to a happy ending :).
